src/components/formControls/radio/style.ts

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License. v. 2.0. If a copy of the MPL was not distributed with this file.
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import styled, { css, ThemedStyledProps } from '../../../theme'
+import { Props, Size } from './index'
+
+interface StyleProps {
+  size?: Size
+  disabled?: boolean
+  selected?: boolean
+}
+
+const getThemeColors = (p: ThemedStyledProps<Props>, selected: boolean | undefined) => {
+  let fillColor = '#696FDC'
+  let borderColor = '#EBECF0'
+
+  if (!p.disabled) {
+    borderColor = selected
+      ? p.theme.color.brandBat
+      : p.theme.color.brandBatActive
+  }
+
+  return css`
+    --radio-fill-color: ${fillColor};
+    --radio-border-color: ${borderColor};
+  `
+}
+
+export const StyledLabel = styled<StyleProps, 'label'>('label')`
+  line-height: 1.3;
+  display: flex;
+  margin-bottom: 20px;
+  color: #686978;
+  font-size: ${p => p.size === 'big' ? '16' : '14'}px;
+  cursor: pointer;
+`
+
+export const StyledInput = styled<{}, 'input'>('input')`
+  display: none;
+`
+
+export const StyledCircle = styled<StyleProps, 'span'>('span')`
+  text-align: center;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-basis: ${p => p.size === 'big' ? '24' : '18'}px;
+  width: ${p => p.size === 'big' ? '24' : '18'}px;
+  height: ${p => p.size === 'big' ? '24' : '18'}px;
+  ${p => getThemeColors(p as ThemedStyledProps<Props>, p.selected)}
+  border: 1px solid var(--radio-border-color);
+  margin-right: ${p => p.size === 'big' ? '17' : '12'}px;
+  border-radius: 50%;
+`
+
+export const StyledFill = styled<StyleProps, 'div'>('div')`
+  width: ${p => p.size === 'big' ? '10' : '8'}px;
+  height: ${p => p.size === 'big' ? '10' : '8'}px;
+  border-radius: 50%;
+  ${p => getThemeColors(p as ThemedStyledProps<Props>, p.selected)}
```

Comment:
May be cleaner to put this function call at the top of this set of rules. Also is the `as ThemeStyledProps<Props>` neccessary? `p` is `ThemedStyleProps<StyleProps>` already - can't the function use that instead?

Reply:
Moved to top of function and `as` typing removed (I had to move some things around to not need to do that conversion)